src/github.com/matrix-org/go-neb/services/guggy/guggy.go

```diff
@@ -0,0 +1,135 @@
+package services
+
+import (
+	"encoding/json"
+	"bytes"
+	"math"
+	log "github.com/Sirupsen/logrus"
+	"github.com/matrix-org/go-neb/matrix"
+	"github.com/matrix-org/go-neb/plugin"
+	"github.com/matrix-org/go-neb/types"
+	"net/http"
+	"strconv"
+	"strings"
+)
+type guggyQuery struct {
+	// "mp4" of "gif"
+	Format string `json:"format"`
+	// Query sentence
+	Sentence  string `json:"sentence"`
+}
+
+type guggyGifResult struct {
+	ReqID string `json:"reqId"`
+	GIF string `json:"gif"`
+	Width float64 `json:"width"`
+	Height float64 `json:"height"`
+}
+
+type guggyService struct {
+	id            string
+	serviceUserID string
+	APIKey        string
+}
+
+func (s *guggyService) ServiceUserID() string { return s.serviceUserID }
+func (s *guggyService) ServiceID() string     { return s.id }
+func (s *guggyService) ServiceType() string   { return "guggy" }
+func (s *guggyService) OnReceiveWebhook(w http.ResponseWriter, req *http.Request, cli *matrix.Client) {
+}
+func (s *guggyService) Register(oldService types.Service, client *matrix.Client) error { return nil }
+func (s *guggyService) PostRegister(oldService types.Service)                          {}
+
+func (s *guggyService) Plugin(client *matrix.Client, roomID string) plugin.Plugin {
+	return plugin.Plugin{
+		Commands: []plugin.Command{
+			plugin.Command{
+				Path: []string{"guggy"},
+				Command: func(roomID, userID string, args []string) (interface{}, error) {
+					return s.cmdGuggy(client, roomID, userID, args)
+				},
+			},
+		},
+	}
+}
+func (s *guggyService) cmdGuggy(client *matrix.Client, roomID, userID string, args []string) (interface{}, error) {
+	// only 1 arg which is the text to search for.
+	querySentence := strings.Join(args, " ")
+	gifResult, err := s.text2gifGuggy(querySentence)
+	if err != nil {
+		return nil, err
+	}
+	mxc, err := client.UploadLink(gifResult.GIF)
```

Comment:
As we found when debugging an earlier problem, if `gifResult.GIF == ""` then this fails with an obscure error message. Can we please check for the empty string and fail more gracefully.